src/main/kotlin/build/buf/intellij/annotator/BufAnalyzeUtils.kt

```diff
@@ -87,7 +91,11 @@ object BufAnalyzeUtils {
         workingDirectory: Path
     ): BufAnalyzeResult? {
         val widget = WriteAction.computeAndWait<BufCLIWidget?, Throwable> {
-            FileDocumentManager.getInstance().saveAllDocuments()
+            FileDocumentManager.getInstance()
+                .saveDocuments { document ->
+                    val psiFile = PsiDocumentManager.getInstance(project).getPsiFile(document) ?: return@saveDocuments false
+                    psiFile is PbFile || BufConfig.CONFIG_FILES.contains(psiFile.name)
+                }
```

Comment:
Updated this to only save documents that impact the buf analysis run (.proto or buf config files).